homeassistant/components/zeroconf.py

```diff
@@ -0,0 +1,49 @@
+"""
+This module exposes Home Assistant via Zeroconf.
+
+Zeroconf is also known as Bonjour, Avahi or Multicast DNS (mDNS).
+
+For more details about Zeroconf, please refer to the documentation at
+https://home-assistant.io/components/zeroconf/
+"""
+import logging
+import socket
+
+from homeassistant.const import (EVENT_HOMEASSISTANT_STOP, __version__)
+
+REQUIREMENTS = ["zeroconf==0.17.5"]
+
+DEPENDENCIES = ["api"]
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = "zeroconf"
+
+ZEROCONF_TYPE = "_home-assistant._tcp.local."
+
+
+def setup(hass, config):
+    """Set up Zeroconf and make Home Assistant discoverable."""
+    from zeroconf import Zeroconf, ServiceInfo
+
+    zeroconf = Zeroconf()
```

Comment:
I tested this by enabling discovery and zeroconf, the two co-existed without issue and the Zeroconf record was published successfully.